### DISCUSSION TITLE
hi, about query item type

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#1035] [Q&A] [answered]
hi, after we converted dgn to bim, we tried to use itwin.js to query the item type data in dgn. Because the user uses Chinese, etc., the ECClass generated by the item type is an unpredictable name, which makes it impossible to use classfullname to accurately find the data.
Now I use some special methods to accurately query the ECClass name of the specified item type, but I always feel that it is not beautiful enough or can not stand the test of practice.

I would like to ask, is there a function similar to ECNameValidation in MicroStation SDK in itwins.js? In order to accurately query item type data that has been used in Chinese, etc., thank you.

**Comment by pmconne** [comment, marked as answer]
The only thing I can find is [ECName](https://www.itwinjs.org/bis/ec/ec-name/?term=ecname), which can only tell you if a name is valid for EC. I don't see any API for encoding/decoding.

@ColinKerr @christophermlawson @rschili

**Reply by ColinKerr** [reply]
I assume you have the name originally entered by the user?  If so you can find the class name via it's display label using ECSql.  Here is an example:
```SQL
SELECT Name, DisplayLabel FROM ECDbMeta.ECClassDef WHERE DisplayLabel = 'Class Display label'
```

If you do not know the original name but want to find a mapping between the names and the entered display labels you can use the same query from above with the WHERE clause removed.

Does this do what you need?  If not can you give a little bit more detail about the parts of the ECNameValidation API that you need.

**Reply by shenshengyi** [reply]
Thank you very much, your answer has solved my problem.
